app/scripts/controllers/membership.js

```diff
@@ -0,0 +1,304 @@
+              var redirectToProjectList = null;
+              var modalScope = createModalScope(subjectName, kindName, roleName, user.metadata.name);
+              if(_.isEqual(subjectName, user.metadata.name)) {
+                if(MembershipService.isLastRole(user.metadata.name, $scope.roleBindings)) {
+                  redirectToProjectList = true;
+                }
+              }
+              $uibModal.open({
+                animation: true,
+                templateUrl: 'views/modals/confirm.html',
+                controller: 'ConfirmModalController',
+                resolve: {
+                  modalConfig: function() {
+                    return modalScope;
+                  }
+                }
+              }).result.then(function() {
+                RoleBindingsService
+                  .removeSubject(subjectName, roleName, $scope.roleBindings, requestContext)
+                  .then(function() {
+                    if(redirectToProjectList) {
+                      $location.url("./");
+                    } else {
+                      refreshRoleBindingList();
+                      showAlert('rolebindingUpdate', 'success', messages.remove.success({
+                        roleName: roleName,
+                        subjectName: subjectName
+                      }));
+                    }
+                  }, function(err) {
+                    showAlert('rolebindingUpdateFail', 'error', messages.remove.error({
+                      roleName: roleName,
+                      subjectName: subjectName
+                    }),  messages.errorReason({
+                      httpErr: $filter('getErrorDetails')(err)
+                    }));
+                  });
+              });
+            },
+            addRoleTo:function(subjectName, kind, roleName, namespace) {
+              var subject = {
+                name: subjectName,
+                kind: kind,
+                namespace: namespace
+              };
+
+              var found = RoleBindingsService.findFirstByRole(roleName, $scope.roleBindings);
+              if(found) {
+                return updateBinding(found, subject, projectName);
+              }
+              return createBinding(roleName, subject, projectName);
+            }
+          });
+
+          // both clusterRoles and local roles are needed, though local are rare
+          RolesService
+            .listAllRoles(requestContext, {
+              // errorNotification: false TODO: DataService.list() does not support
+            })
+            .then(function(resp) {
+              allRoles = MembershipService.mapRolesForUI(_.first(resp), _.last(resp));
+              var sortedRoles = MembershipService.sortRoles(allRoles);
+              var filteredRoles = MembershipService.filterRoles(allRoles);
+              var includesRole = function(roleName, roles) {
+                return _.includes(_.pluck(roles, 'metadata.name'), roleName);
```

Comment:
This is effectively iterating over the list of roles twice.  Either of these will do it in a one-liner still and should do a single iteration.

```
_.some(roles, { metadata: { name: roleName } });
```

or alternatively

```
_.some(roles, _.matchesProperty('metadata.name', roleName));
```